### PR TITLE
sent users to /settings/plan on all upsells with "standard" plan selected

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @cvializ

### CODE REVIEW

**Comment by andrewconner** [comment]
Seems reasonable :)

File: kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js [hunk 1]
```diff
@@ -5,12 +5,13 @@ angular.module('kifi')
 .controller('PaymentPlanCtrl', [
   '$window', '$rootScope', '$scope', '$state', '$filter', '$q',
   'billingState', 'billingService', 'modalService', 'StripeCheckout',
-  'messageTicker', 'paymentPlans', '$timeout',
+  'messageTicker', 'paymentPlans', '$timeout', 'upgrade', '$log',
   function ($window, $rootScope, $scope, $state, $filter, $q,
             billingState, billingService, modalService, StripeCheckout,
-            messageTicker, paymentPlans, $timeout) {
+            messageTicker, paymentPlans, $timeout, upgrade, $log) {
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js]: ruh roh

File: kifi-backend/modules/shoebox/angular/src/route.js [hunk 2]
```diff
@@ -184,6 +184,12 @@ angular.module('kifi')
               return billingService
               .getBillingPlans(profile.organization.id);
             }
+          ],
+          upgrade: [
+            '$state', '$stateParams',
+            function ($state, $stateParams) {
+              return !!$stateParams.upgrade;
+            }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/route.js]: removing this... it's equivalent to calling $state.params.upgrade from the controller

File: kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js [hunk 3]
```diff
@@ -103,11 +104,21 @@ angular.module('kifi')
...
+        cycle: standardTierPlans[0].cycle,
+        newCard: null
+      };
+    } else {
+      $scope.plan = {
+        name: currentPlan.name,
+        cycle: currentPlan.cycle, //months
+        newCard: null
+      };
+    }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js:121]: @cvializ

File: kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ angular.module('kifi')
             messageTicker, paymentPlans, $timeout) {
     $scope.billingState = billingState;
     $scope.card = billingState.card;
+    $scope.upgrade = !!$state.params.upgrade;
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js:14]: @cvializ

File: kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js (hunk 3 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js:121]: Your approach looks good, but I believe that Product wants the annual cycle to be the default, and I think `standardTierPlans[0]` might be the monthly cycle.

File: kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js (hunk 4 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js:14]: nice, works with the ?querystring and the boolean value from the `ui-sref`s

**Comment by cvializ** [comment]
:+1: hooray! Just make sure it selects the annual plan by default

File: kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js (hunk 3 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/paymentPlan.js:121]: good call, thanks

**Pull request merged**